src/ZkFold/Symbolic/Data/ByteString.hs

```diff
@@ -250,39 +180,71 @@ instance
     ) => ReverseEndianness wordSize (ByteString n (Interpreter (Zp p))) where
     reverseEndianness (ByteString (Interpreter v)) = ByteString . Interpreter $ reverseEndianness' @wordSize v
 
-instance (Finite (Zp p), KnownNat n) => BoolType (ByteString n (Interpreter (Zp p))) where
-    false = fromConstant (0 :: Natural)
 
-    -- | A ByteString with all bits set to 1 is the unity for bitwise and.
+instance (Symbolic c, KnownNat n) => BoolType (ByteString n c) where
+    false = fromConstant (0 :: Natural)
     true = not false
 
-    -- | bitwise not.
-    -- @Data.Bits.complement@ is undefined for @Natural@, we have to flip bits manually.
-    not = fromConstant @Natural . (nextPow2 -!) . toConstant
-      where
-        nextPow2 :: Natural
-        nextPow2 = (2 ^ natVal (Proxy @n)) -! 1
-
-    -- | Bitwise or
-    x || y = fromConstant @Natural $ toConstant x .|. toConstant y
+    not (ByteString bits) = ByteString $ symbolicF bits (V.parFmap (one -) ) solve
```

Comment:
Nice!
This can be even done with `fromCircuit` as computations in both contexts are tautological